### PR TITLE
Throw error on add_forecast for unsupported component

### PR DESCRIPTION
**Pull request by daniel-thom**

### CODE REVIEW

File: src/component.jl [hunk 1]
```diff
@@ -2,7 +2,12 @@ function add_forecast!(
                        component::T,
                        forecast::ForecastInternal,
                       ) where T <: InfrastructureSystemsType
-    add_forecast!(_get_forecast_container(component), forecast)
+    container = _get_forecast_container(component)
+    if isnothing(container)
+        throw(ArgumentError("type $T does not contain forecasts"))
```
Comment (jd-lara) [inline src/component.jl]: The message isn't clear if the cause of the error is the type $T not having the forecast field or the data not being available. 

I would suggest 
```Julia
throw(ArgumentError("type $T can not store forecasts"))
```

Comment (daniel-thom) [reply src/component.jl]: Fixed

File: src/component.jl [hunk 2]
```diff
@@ -4,7 +4,7 @@ function add_forecast!(
                       ) where T <: InfrastructureSystemsType
     container = _get_forecast_container(component)
     if isnothing(container)
-        throw(ArgumentError("type $T does not contain forecasts"))
+        throw(ArgumentError("type $T cannot store forecasts"))
```
Comment (jd-lara) [inline src/component.jl]: ```throw(ArgumentError("type $T does not support storing forecasts"))``` ?

Comment (daniel-thom) [reply src/component.jl]: That's better.  Fixed.

**Pull request merged**